File: Undertale_Website/js/mainChar.js
const apiUrl = 'http://localhost:6969/char/mainChar';

const getMainCharacters = async () => {
  try {
    const response = await fetch(apiUrl);
    const data = await response.json();

    const charactersContainer = document.getElementById('character-card');
    let charactersHTML = '';

    data.forEach(character => {
        console.log(character.name);
        const card = createCharacterCard(character);
        charactersHTML += card;
    });
    charactersContainer.innerHTML = charactersHTML;
  } catch (error) {
    console.error(error);
  }
};

const createCharacterCard = (character) => {

    console.log(character);
    var text = 
        `<div class="card">
            <h2 class="card-title">${character.name}</h2>
            <div class="card-info">
                <p><u>Appearances:</u> <span class="appearances"><br>${character.appearances.join(', ')}</span></p>
                <p><u>Role:</u> <span class="role"><br>${character.role}</span></p>
                <p><u>Status:</u> <span class="status"><br>${character.status}</span></p>
                <p><u>Max Health:</u> <span class="max-health"><br>${character.maxHealth}</span></p>
                <h3>Abilities:</h3>
                <ul class="abilities">          
    `;
        
    for (let i in character.abilities) {
        text += `<li><u><span class="ability-${i+1}-name">${character.abilities[i].name}</span>:</u> <span class="ability-${i+1}-features"><br>${character.abilities[i].features.join(', ')}</span></li>`;
    }

    text +=           
            `</ul>
        </div>
    </div>`
    return text;       
}

getMainCharacters();

File: Undertale_WPF/Undertale_WPF/MainChar.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Undertale_WPF
{
    public class MainChar : Interface1
    {
        public string Id { get; set; }
        [JsonProperty("name")] // specify actual name for MongoDB
        public string Name { get; set; }
        [JsonProperty("appearances")] // specify actual name for MongoDB
        public List<string> Appearances { get; set; }
        [JsonProperty("role")] // specify actual name for MongoDB
        public string Role { get; set; }
        [JsonProperty("status")] // specify actual name for MongoDB
        public string Status { get; set; }
        [JsonProperty("maxHealth")] // specify actual name for MongoDB
        public int MaxHealth { get; set; }
        [JsonProperty("abilities")] // specify actual name for MongoDB
        public Dictionary<string, Ability> Abilities { get; set; }
        
        public MainChar()
        {

        }

        public MainChar(string? id, string name, List<string> appearances, string role, string status, int maxHealth, Dictionary<string, Ability> abilities)
        {
            this.Id = id;
            this.Name = name;
            this.Appearances = appearances;
            this.Role = role;
            this.Status = status;
            this.MaxHealth = maxHealth;
            this.Abilities = abilities;
        }

        public class Ability
        {
            [JsonProperty("name")] // specify actual name for MongoDB
            public string Name { get; set; }
            [JsonProperty("features")] // specify actual name for MongoDB
            public List<string> Features { get; set; }

            public Ability (string name, List<string> features)
            {
                this.Name = name;
                this.Features = features;
            }
        } 
    }
}
